lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -213,6 +214,14 @@ private void flush()
         }
         List<BufferData> bufferDataList = builder.build();
 
+        if (rows == 0) {
+            // Avoid writing empty row groups as these are ignored by the reader
+            verify(
```

Comment:
Is it possible to write a test?

Reply:
This is going to get tested via https://github.com/trinodb/trino/pull/13246 (validation fails because reader ignores empty row group and counts 1 fewer row group than what the writer wrote), that's how I noticed it.